### DISCUSSION TITLE
How does the content-based duplicate detection work?

### DISCUSSION THREAD

**Discussion by quasimik** [firefly-iii/firefly-iii#6126] [Firefly III]
I have two asset accounts, A and B, and I want to import two separate CSV statements into each of them using Data Importer (FIDI). In each statement, there is a transfer made from A to B of $234.56, but each statement records this under a different description. For example:

Statement A:
```
Date,Amount,Description
...
2022-03-12,-234.56,"TRF OUT ACC B"
...
```

Statement B:
```
Date,Amount,Description
...
2022-03-12,234.56,"Incoming deposit"
...
```

I would first import all of Statement A into Account A, and correctly wire up the transfer transaction. As such, the transfer would already appear under Account B with the description "TRF OUT ACC B". Then, I want to import all of Statement B into Account B, but I don't want to record the duplicate transfer.

Questions.
1. Would firefly's "content-based duplicate detection" detect these as duplicates?
2. When importing Statement B, is there a way to combine the incoming description from Statement B's transfer transaction ("Incoming deposit") with the existing description in FF from the earlier import of Statement A's transaction ("TRF OUT ACC B")?

**Comment by JC5** [comment]
https://docs.firefly-iii.org/data-importer/usage/configure/#duplicate-transaction-detection

1. No, they are different, so they will not be recognized as a duplicate. Since your transaction doesn't seem to have a unique identifier, you won't be able to use "identifier-based detection".
2. No. You can use rules to change the description, but not based on another transaction.
